smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -0,0 +1,561 @@
+                .createPacketCollector(new IQReplyFilter(mamQueryIq, connection));
+
+        try {
+            connection.sendStanza(mamQueryIq);
+            mamResponseQueryIQ = mamResponseQueryIQCollector.nextResultOrThrow();
+        } finally {
+            mamResponseQueryIQCollector.cancel();
+        }
+
+        return mamResponseQueryIQ.getDataForm().getFields();
+    }
+
+    /**
+     * Returns true if Message Archive Management is supported by the server.
+     * 
+     * @return true if Message Archive Management is supported by the server.
+     * @throws NotConnectedException
+     * @throws XMPPErrorException
+     * @throws NoResponseException
+     * @throws InterruptedException
+     */
+    public boolean isSupportedByServer()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return ServiceDiscoveryManager.getInstanceFor(connection()).serverSupportsFeature(MamElements.NAMESPACE);
+    }
+
+    private DataForm getNewMamForm() {
+        FormField field = new FormField(FormField.FORM_TYPE);
+        field.setType(FormField.Type.hidden);
+        field.addValue(MamElements.NAMESPACE);
+        DataForm form = new DataForm(DataForm.Type.submit);
+        form.addField(field);
+        return form;
+    }
+
+    /**
+     * Get the preferences stored in the server.
+     * 
+     * @return the MAM preferences result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamPrefsResult retrieveArchivingPreferences() throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, NotLoggedInException {
+        MamPrefsIQ mamPrefIQ = prepareRetrievePreferencesStanza();
+        return queryMamPrefs(mamPrefIQ);
+    }
+
+    private MamPrefsIQ prepareRetrievePreferencesStanza() {
+        MamPrefsIQ mamPrefIQ = new MamPrefsIQ(Type.get, null, null, null);
+        return mamPrefIQ;
+    }
+
+    /**
+     * Update the preferences in the server.
+     * 
+     * @param alwaysJids
+     *            is the list of JIDs that should always have messages to/from
+     *            archived in the user's store
+     * @param neverJids
+     *            is the list of JIDs that should never have messages to/from
+     *            archived in the user's store
+     * @param defaultField
+     *            can be "roster", "always", "never" (look at the XEP-0313
+     *            documentation)
+     * @return the MAM preferences result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamPrefsResult updateArchivingPreferences(List<Jid> alwaysJids, List<Jid> neverJids, String defaultField)
```

Comment:
defaultField should have been an enum. Never give the user a chance to use false values.